driver-core/src/main/com/mongodb/event/EventListenerSettings.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.event;
+
+import com.mongodb.annotations.NotThreadSafe;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.mongodb.assertions.Assertions.notNull;
+import static java.util.Collections.unmodifiableList;
+
+/**
+ * Settings for the various event listeners.
+ *
+ * @since 3.5
+ */
+public final class EventListenerSettings {
```

Comment:
I just don't see that EventListenerSettings simplifies anything for users more than associating listeners with the existing settings classes (e.g. ClusterSettings).  Scattering listeners amongst settings classes is not problematic in my view, so long as there is a sensible home for each.  And it avoids a bunch of deprecations for the ones already there.

I'm not concerned about ClusterFactory.  Even without the listeners that interface was a failed experiment.

If you still feel this is the right way forward, I'll accept it.  Consider this objection advisory.  Ultimately it's a judgement call and there's no one right answer.